core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java

```diff
@@ -195,18 +192,107 @@ private static void callAuthorizationPluginImpl(
   }
 
   public static boolean needApplyAuthorizationPluginAllCatalogs(SecurableObject securableObject) {
-    // TODO: Add `supportsSecurableObjects` method for every privilege to simplify this code
     if (securableObject.type() == MetadataObject.Type.METALAKE) {
       List<Privilege> privileges = securableObject.privileges();
       for (Privilege privilege : privileges) {
-        if (!pluginNotSupportsPrivileges.contains(privilege.name())) {
+        if (privilege.supportsMetadataObjectType(MetadataObject.Type.CATALOG)) {
           return true;
         }
       }
     }
     return false;
   }
 
+  // Check every securable object whether exists and is imported.
+  public static void checkSecurableObject(String metalake, MetadataObject object) {
+    NameIdentifier identifier = MetadataObjectUtil.toEntityIdent(metalake, object);
+
+    Supplier<NoSuchMetadataObjectException> exceptionToThrowSupplier =
+        () ->
+            new NoSuchMetadataObjectException(
+                "Securable object %s doesn't exist", object.fullName());
+
+    switch (object.type()) {
+      case METALAKE:
+        check(
+            GravitinoEnv.getInstance().metalakeDispatcher().metalakeExists(identifier),
```

Comment:
It is not good to use global static env variable here directly, which makes you hard to do the UT.